src/postgres-client/src/error.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! [PostgresClient]-related errors.
+
+use std::fmt;
+
+/// An error coming from Postgres.
+#[derive(Debug)]
+pub enum PostgresError {
+    /// A determinate error from Postgres.
+    Determinate(anyhow::Error),
```

Comment:
Will the oracle need to treat determinate and indeterminate errors differently? If not, I probably wouldn't have bothered with this, but it also seems fine as you have it.

Reply:
I think it will not care, and I largely had this to mirror the existing errors in persist/consensus. My current thinking is that the oracle will handle all retries internally and we don't care about the difference between determinate/indeterminate because all operations can always be retried (we might burn a few write timestamps, that's all).